nodes/text/viewer_text_mk3.py

```diff
@@ -0,0 +1,226 @@
+            if insert.is_linked:
+                label = insert.other.node.label
+                if label:
+                    label = '; node ' + label.upper()
+                name = insert.name.upper()
+                # vertices socket
+                if insert.other.bl_idname == 'VerticesSocket':
+                    itype = '\n\nSocket ' + name + label + '; type VERTICES: \n'
+
+                # edges/faces socket
+                elif insert.other.bl_idname == 'StringsSocket':
+                    itype = '\n\nSocket ' + name + label + '; type EDGES/POLYGONS: \n'
+
+                # matrix socket
+                elif insert.other.bl_idname == 'MatrixSocket':
+                    itype = '\n\nSocket ' + name + label + '; type MATRICES: \n'
+
+                # object socket
+                elif insert.other.bl_idname == 'SvObjectSocket':
+                    itype = '\n\nSocket ' + name + label + '; type OBJECTS: \n'
+                # else
+                else:
+                    itype = '\n\nSocket ' + name + label + '; type DATA: \n'
+
+                eva = insert.sv_get()
+                deptl = levelsOflist(eva)
+                if deptl and deptl > 2:
+                    a = self.readFORviewer_sockets_data(eva, deptl, len(eva))
+                elif deptl:
+                    a = self.readFORviewer_sockets_data_small(eva, deptl, len(eva))
+                else:
+                    a = 'None'
+                outs += itype+str(a)+'\n'
+        self.do_text(outs,node)
+        return {'FINISHED'}
+
+    def makeframe(self, nTree):
+        '''
+        Making frame to show text to user. appears in left corner
+        Todo - make more organized layout with button making
+        lines in up and between Frame and nodes and text of user and layout name
+        '''
+        # labls = [n.label for n in nTree.nodes]
+        if any('Sverchok_viewer' == n.label for n in nTree.nodes):
+            return
+        else:
+            a = nTree.nodes.new('NodeFrame')
+            a.width = 800
+            a.height = 1500
+            locx = [n.location[0] for n in nTree.nodes]
+            locy = [n.location[1] for n in nTree.nodes]
+            mx, my = min(locx), max(locy)
+            a.location[0] = mx - a.width - 10
+            a.location[1] = my
+            a.text = bpy.data.texts['Sverchok_viewer']
+            a.label = 'Sverchok_viewer'
+            a.shrink = False
+            a.use_custom_color = True
+            # this trick allows us to negative color, so user accept it as grey!!!
+            color = [1 - i for i in bpy.context.user_preferences.themes['Default'].node_editor.space.back[:]]
+            a.color[:] = color
+
+    def do_text(self,outs,node):
+        nTree = bpy.data.node_groups[self.treename]
+        #this part can be than removed from node text viewer:
+
+        if not 'Sverchok_viewer' in bpy.data.texts:
+            bpy.data.texts.new('Sverchok_viewer')
+
+        podpis = '\n'*2 \
```

Comment:
Huh. At least some of Sverchok developers are not russian, so let's rename this to "footer", if you do not mind :)